### PR TITLE
py-astrovascpy: new BlueBrain Python package

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py:1]: All Python dependencies should be `"build", "run"`, unless they are pure install ([`wheel` is not needed](https://github.com/BlueBrain/spack/blob/develop/lib/spack/spack/build_systems/python.py#L201)).

File: bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    depends_on("py-coverage", type="build")
+    depends_on("py-cython", type="build")
+    depends_on("py-h5py", type="build")
+    depends_on("py-libsonata", type="build")
+    depends_on("py-matplotlib", type="build")
+    depends_on("py-morphio", type="build")
+    depends_on("py-networkx", type="build")
+    depends_on("py-numpy", type="build")
+    depends_on("py-pandas", type="build")
+    depends_on("petsc", type="build")
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]: Should just be defaulted type-wise

File: bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    depends_on("py-coverage", type=("build", "run"))
+    depends_on("py-cython", type=("build", "run"))
+    depends_on("py-h5py", type=("build", "run"))
+    depends_on("py-libsonata", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+    depends_on("py-morphio", type=("build", "run"))
+    depends_on("py-networkx", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-pandas", type=("build", "run"))
+    depends_on("petsc", type=("build"))
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]:
```suggestion
    depends_on("petsc")
```
`petsc` does not look like a header-only library.

Comment (tristan0x) [reply bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]: fixed in [6394ff3](https://github.com/BlueBrain/spack/pull/2175/commits/6394ff3ea1348e15563b27ed93c354f9f96e687a)

File: bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py (hunk 1 above)
Comment (tristan0x) [reply bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]: fixed in [2eda903](https://github.com/BlueBrain/spack/pull/2175/commits/2eda9035d080d19a8b3df2ed08cfb24c27bf3a2a)

**Review by matz-e — DISMISSED** [review]

File: bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    url = "https://github.com/BlueBrain/AstroVascPy/releases/tag/0.1"
+
+    maintainers("tristan0x")
+
+    version("0.1", sha256="1e9b2036e9c1ce4cb0650612c0be2e7e7b142754e1550dfbfb0b757cee4f8bdd")
+
+    variant("vtk", default=False, description="add VTK support (mainly for visualization)")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools-scm", type=("build", "run"))
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]:
```suggestion
    depends_on("py-setuptools-scm", type="build")
```
I recall this being a build dependency mostly (just saw that when I hit approve :facepalm:)

Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-astrovascpy/package.py]: Sorry, that comment may be contradictory to the other one. Pure Python build dependencies should really be `build` only.

**Review by matz-e — APPROVED** [review]

**Pull request merged**